### PR TITLE
Pandaren faction choose

### PR DESCRIPTION
**Pull request by Hyjarion**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

-  Opcode for the pandaren faction choose UI menu
-  A new player function that shows the menu
-  Script for Spirit of Master Shang Xi (At "A New Fate" quest)

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Tests performed:** (Does it build, tested in-game, etc.)
Built and tested in-game

### CODE REVIEW

**Comment by Shauren** [comment]
what is this atrocious skyfire port?

File: src/server/game/Handlers/MiscHandler.cpp [hunk 1]
```diff
@@ -1173,3 +1173,43 @@ void WorldSession::HandleMountSetFavorite(WorldPackets::Misc::MountSetFavorite&
 {
     _collectionMgr->MountSetFavorite(mountSetFavorite.MountSpellID, mountSetFavorite.IsFavorite);
 }
+
+#define JOIN_THE_ALLIANCE 1
+#define JOIN_THE_HORDE    0
```
Comment (tkrokli) [inline src/server/game/Handlers/MiscHandler.cpp]: Notice that TC uses `enum` for simple values like this, in ascending order.

File: src/server/game/Entities/Player/Player.h [hunk 2]
```diff
@@ -2515,6 +2515,8 @@ class TC_GAME_API Player : public Unit, public GridObject<Player>
 
         SceneMgr& GetSceneMgr() { return m_sceneMgr; }
 
+		void ShowNeutralPlayerFactionSelectUI();
```
Comment (jackpoz) [inline src/server/game/Entities/Player/Player.h]: tabs

File: src/server/scripts/World/npcs_special.cpp [hunk 3]
```diff
@@ -2606,6 +2606,40 @@ class npc_argent_squire_gruntling : public CreatureScript
...
+
+	bool OnGossipHello(Player* player, Creature* npc)
+	{
+		if (npc->IsQuestGiver())
+			player->PrepareQuestMenu(npc->GetGUID());
+
+		if (player->getRace() == RACE_PANDAREN_NEUTRAL)
+		{
+			if (player->GetQuestStatus(31450) == QUEST_STATUS_INCOMPLETE)
+				AddGossipItemFor(player, 0, "I'm ready to decide.", GOSSIP_SENDER_MAIN, 1);
```
Comment (jackpoz) [inline src/server/scripts/World/npcs_special.cpp]: hardcoded text

File: src/server/scripts/World/npcs_special.cpp [hunk 4]
```diff
@@ -2606,6 +2606,40 @@ class npc_argent_squire_gruntling : public CreatureScript
...
+	npc_Spirit_of_Master_Shang_Xi() : CreatureScript("npc_Spirit_of_Master_Shang_Xi") { }
+
+	bool OnGossipHello(Player* player, Creature* npc)
+	{
+		if (npc->IsQuestGiver())
+			player->PrepareQuestMenu(npc->GetGUID());
+
+		if (player->getRace() == RACE_PANDAREN_NEUTRAL)
+		{
+			if (player->GetQuestStatus(31450) == QUEST_STATUS_INCOMPLETE)
```
Comment (jackpoz) [inline src/server/scripts/World/npcs_special.cpp]: magic number

File: src/server/scripts/World/npcs_special.cpp [hunk 5]
```diff
@@ -2606,6 +2606,40 @@ class npc_argent_squire_gruntling : public CreatureScript
...
+		SendGossipMenuFor(player, DEFAULT_GOSSIP_MESSAGE, npc);
+
+		return true;
+	}
+
+	bool OnGossipSelect(Player* player, Creature* npc, uint32 sender, uint32 uiAction)
+	{
+
+		switch (uiAction)
+			case 1:
```
Comment (jackpoz) [inline src/server/scripts/World/npcs_special.cpp]: first time in my life I see a switch without brackets

**Comment by jackpoz** [comment]
travis build reported as broken, please review the warnings/errors

**Comment by Aokromes** [comment]
[ 85%] Building CXX object src/server/scripts/CMakeFiles/scripts_world.dir/World/item_scripts.cpp.o
/home/travis/build/TrinityCore/TrinityCore/src/server/scripts/World/npcs_special.cpp:2630:48: fatal error: 
      unused parameter 'npc' [-Wunused-parameter]
        bool OnGossipSelect(Player* player, Creature* npc, uint32 sender...
                                                      ^

**Comment by Lopfest** [comment]
@TheHighlord, i will use HandleSelectFactionOpcode and ShowNeutralPlayerFactionSelectUI in my [branch](https://github.com/TrinityCore/TrinityCore/pull/17021) if you don't mind :)

**Pull request closed without merging**